src/module_3/push_exploration.md

```diff
@@ -0,0 +1,5364 @@
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>5</th>
+      <td>33826472919172</td>
+      <td>ricepastapulses</td>
+      <td>2808434524292</td>
+      <td>3479090790532</td>
+      <td>2020-10-06 10:50:23</td>
+      <td>2020-10-06 00:00:00</td>
+      <td>3</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.081052</td>
+      <td>0.053512</td>
+      <td>clearspring</td>
+      <td>0.038462</td>
+      <td>2.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+```python
+# We will remeber the classification we did in previous notebook:
+
+predicted = ['outcome']
+information = ['variant_id', 'order_id', 'user_id', 'created_at', 'order_date']
+numerical = ['user_order_seq', 'normalised_price', 'discount_pct', 'global_popularity',
+            'count_adults', 'count_children', 'count_babies', 'count_pets', 
+            'people_ex_baby', 'days_since_purchase_variant_id', 
+            'avg_days_to_buy_variant_id', 'std_days_to_buy_variant_id',
+            'days_since_purchase_product_type', 'avg_days_to_buy_product_type',
+                'std_days_to_buy_product_type']
+
+categorical = ['product_type', 'vendor']
+binary = ['ordered_before', 'abandoned_before', 'active_snoozed', 'set_as_regular']
+```
+
+From numerical: 
+- We will remove count_adults, count_children, count_pets and keep only count_adults, which seems to be highly representative. We will maintain count_babies as correlation it is not that high (0.15) and may provide info.
+- We will remove std_days_to_buy_product_type and keep avg_days_to_buy_product_type (highly correlated between themselves. We may do it reversely also)
+- We will remove std_days_to_buy_variant_id and keep avg_days_to_buy_variant_id (same thing)
+
+From categorical:
+- We will remove vendor and keep product_type (the former has too different values). Later we will apply one-hot encoding
+
+From binary: 
+- We would try to use some resampling technique for grouping the 4 into just 1. We will create a column 'any_event' to input value '1' if any of the four cols has a value of 1. In this sense, we address the unbalanced distribution (a bit), we simplify it and also keep the info if any of these events occured
+
+
+
+```python
+# Remove numericals
+numerical_remove = ['count_adults', 'count_children', 'count_babies',
+                    'std_days_to_buy_product_type', 'std_days_to_buy_variant_id']
+numerical = [col for col in numerical if col not in numerical_remove]
+
+# Remove categoricals
+categorical.remove('vendor')
```

Comment:
not a big deal, but if with the numerical columns you are creating a vector `numerical_remove`, why don't do the same with the categorical ones